packages/appkit/src/adapters/ChainAdapterBlueprint.ts

```diff
@@ -455,6 +479,224 @@ export abstract class AdapterBlueprint<
 
     return connector
   }
+
+  /**
+   * Handles connect event for a specific connector.
+   * @param {string[]} accounts - The accounts that changed
+   * @param {string} connectorId - The ID of the connector
+   */
+  protected onConnect(accounts: (ParsedCaipAddress | string)[], connectorId: string) {
+    if (accounts.length > 0) {
+      const account = accounts[0]
+      const address = typeof account === 'string' ? account : account?.address
+      const chainId = typeof account === 'string' ? undefined : account?.chainId
+
+      const caipNetwork = this.getCaipNetworks()
+        .filter(n => n.chainNamespace === this.namespace)
+        .find(n => n.id.toString() === chainId?.toString())
+
+      const connector = this.connectors.find(c => c.id === connectorId)
+
+      if (address) {
+        this.emit('accountChanged', {
+          address,
+          chainId,
+          connector
+        })
+
+        this.addConnection({
+          connectorId,
+          accounts: accounts.map(_account => {
+            if (typeof _account === 'string') {
+              return { address: _account }
+            }
+
+            return { address: _account.address }
```

Comment:
same. 

can we make it a util? maybe it's responsibility of the specific namespace adapter to parse it to a bare addresS?